test/mock_high_priority_service.cpp

```diff
@@ -69,14 +82,29 @@ int wsrep::mock_high_priority_service::rollback(
 }
 
 int wsrep::mock_high_priority_service::apply_toi(const wsrep::ws_meta&,
-                                                 const wsrep::const_buffer&)
+                                                 const wsrep::const_buffer&,
+                                                 wsrep::mutable_buffer&)
 {
     assert(client_state_->transaction().active() == false);
     assert(client_state_->toi_meta().seqno().is_undefined() == false);
     return (fail_next_toi_ ? 1 : 0);
 }
 
+void wsrep::mock_high_priority_service::adopt_apply_error(
+    const wsrep::mutable_buffer& err)
+{
+    client_state_->adopt_apply_error(err);
+}
+
 void wsrep::mock_high_priority_service::after_apply()
 {
     client_state_->after_applying();
 }
+
+int wsrep::mock_high_priority_service::log_dummy_write_set(
+    const wsrep::ws_handle&,
+    const wsrep::ws_meta&,
+    const wsrep::mutable_buffer& err)
+{
+    return err.size() > 0;
```

Comment:
Why is non-zero error buffer considerd error here? I'd assume that successfully logging a dummy write set would return success even if there is an error buffer to be logged.

Reply:
> I'd assume that successfully logging a dummy write set would return success even if there is an error buffer to be logged.
No, why? There will be commit monitor release and voting. So error can be an error

Reply:
Is monitor release and voting really supposed to happen inside this call? Couldn't it be done by the caller to reduce the responsibilities of `log_dummy_write_set()` implementation?

Reply:
Ok, I see it now. The commit monitor release must happen inside `log_dummy_write_set()` to support group commit, which implies that also voting has to happen inside.